File: src/main/java/com/eugene/java/controller/HomeController.java
package com.eugene.java.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HomeController {

  /**
   * request URL: <a href="http://localhost:8080/url-encoded/wang%20eugene">...</a> response: Hello,
   * wang eugene!
   */
  @GetMapping("/url-encoded/{name}")
  public String urlEncoded(@PathVariable String name) {
    return "Hello, " + name + "!";
  }
}

File: fart/src/main/kotlin/com/eugene/fart/controller/HomeController.kt
package com.eugene.fart.controller

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController("/")
class HomeController {
    @GetMapping("/")
    fun home(): String {
        return "Welcome to Fart!"
    }
}